_drafts/conviva-dag-process-time-prod-issue-draft.md

```diff
@@ -77,9 +77,17 @@ pub struct TypeRegistry<C: Clone, const N: usize = DEFAULT_N> {
 
 ## Post Mortem
 
-So, ArcSwap finally fixed the problem but it is worthwhile to understand why exactly that worked where concurrent hash maps had failed. It's worth repeating here again that our type registry, which is where the hash map was being used, was an *almost* read-only scenario.
+First, let's look a little more closely at how some concurrent hash maps typically operate. Many designs involve mechanisms like counters to track readers and writers, though the specifics can vary. For example, some implementations use a single, shared counter (not unlike a typical `RWLock`), while others employ sharded designs or multiple counters to reduce contention.
```

Comment:
Thanks, I think this helps a lot.  No need to even clarify with DashMap details below if you feel that's too much